proposals/0xxx-enum-cases-as-protocol-witnesses.md

```diff
@@ -0,0 +1,211 @@
+
+```swift
+enum Foo {
+  case bar(_ value: Int)
+  case baz
+}
+
+let f = Foo.bar // `f` is a function of type (Int) -> Foo
+let bar = f(2) // Returns Foo
+let baz = Foo.baz // Returns Foo
+```
+
+is the same as:
+
+```swift
+struct Foo {
+  static func bar(_ value: Int) -> Self { ... }
+  static var baz: Self { ... }
+}
+
+let f = Foo.bar // `f` is a function of type (Int) -> Foo
+let bar = f(2) // Returns Foo
+let baz = Foo.baz // Returns Foo
+```
+
+Such "spelling" exceptions exist when matching other kinds of requirements as well, for example:
+
+```swift
+protocol Foo {
+  var somePropertyA: Self { get }
+}
+
+struct ImplementsFoo: Foo {
+  // This can be a 'let' because even though the
+  // keywords don't match, and a variable and a 
+  // constant are two different things, the
+  // *semantics* of 'var ... { get }' and 'let'
+  // do match.
+  let somePropertyA: Self
+  // and you can write it as a 'var' if you want 
+  // and still keep the semantics the same.
+  var somePropertyA: Self
+}
+```
+
+Now, because enum cases are not considered as a "witness" for static protocol requirements, one has to provide a manual implementation instead:
+
+```swift
+enum JSONDecodingError: DecodingError {
+  case _fileCorrupted
+  case _keyNotFound(_ key: String)
+  static var fileCorrupted: Self { return ._fileCorrupted }
+  static func keyNotFound(_ key: String) -> Self { 
+    return ._keyNotFound(key) 
+  }
+}
+```
+
+This leads to some rather unfortunate consequences:
+
+1. The ability to write a case with the same name as the requirement is lost. Now, you can rename the case to something different, but it might not always be ideal, especially because naming things right is a really hard problem. In most cases, you expect the case to be named the same as the requirement.
+2. The namespace of the enum is now polluted with both cases and requirements (for example, in the snippet above we have `_fileCorrupted` and `fileCorrupted`), which can be confusing during code completion.
+3. There's extra code that now has to be maintained and which arguably should not exist in the first place.
+
+In almost every corner of the language, enum cases and static properties/functions are indistinguishable from each other, *except* when it comes to matching protocol requirements, which is very inconsistent, so it is not unreasonable to think of a enum case without associated values as a `static`, get-only property that returns `Self` or an enum case with associated values as a `static` function (with arguments) that returns `Self`.
+
+Doing so can also lead to other improvements, for example, one can conform `DispatchTimeInterval` direcly to Combine's `SchedulerTypeIntervalConvertible` instead of having to go through a much more complicated type like `DispatchQueue.SchedulerTimeType.Stride`:
```

Comment:
`direcly` ➡️ `directly`

```suggestion
Doing so can also lead to other improvements, for example, one can conform `DispatchTimeInterval` directly to Combine's `SchedulerTypeIntervalConvertible` instead of having to go through a much more complicated type like `DispatchQueue.SchedulerTimeType.Stride`:
```